tests/BackupRetentionTest.php

```
@@ -420,11 +414,126 @@ public function testInputOrderDoesNotAffectResult(): void
+    /**
+     * keep_*=0 disables that tier entirely. The `if ($keep* > 0)` guards mean
+     * a zero-tier produces no slot tracking and no keepers from that tier.
+     * The newest backup is still preserved by the safety guard regardless.
+     *
+     * Behaviour documented: all-zero config keeps only the newest, prunes the
+     * rest. Mixed (e.g. keep_hourly=0 with daily/weekly/monthly>0) skips
+     * hourly without affecting other tiers' promotions.
+     */
+    public function testKeepZeroDisablesTier(): void
+    {
+        $now = strtotime('2026-04-28 12:00:00');
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$now'. (undefined)

(UnusedLocalVariable)